### PR TITLE
[VimeoIE]Add support to download original file.

### PR DESCRIPTION
**Pull request by Ashish0804**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

closes https://github.com/yt-dlp/yt-dlp/issues/597

### CODE REVIEW

File: yt_dlp/extractor/vimeo.py [hunk 1]
```diff
@@ -253,6 +253,23 @@ def _extract_original_format(self, url, video_id, unlisted_hash=None):
                             'quality': 1,
                         }
 
+    def _extract_original_format_2(self, video_id):
```
Comment (Ashish0804) [inline yt_dlp/extractor/vimeo.py:256]: Name could be better tbh
So open to suggestions.

File: yt_dlp/extractor/vimeo.py [hunk 2]
```diff
@@ -253,6 +253,23 @@ def _extract_original_format(self, url, video_id, unlisted_hash=None):
...
+        jwt_response = self._download_json('https://vimeo.com/_rv/viewer', video_id)
+        if jwt_response.get('jwt'):
+            headers = {'Authorization': f"jwt {jwt_response['jwt']}"}
```
Comment (Ashish0804) [inline yt_dlp/extractor/vimeo.py:259]: What's supposed to be the guidelines for quotes in an f string?
should it be `"` inside of `'` or other way round?

File: yt_dlp/extractor/vimeo.py [hunk 3]
```diff
@@ -253,6 +253,23 @@ def _extract_original_format(self, url, video_id, unlisted_hash=None):
...
+            original_response = self._download_json('https://api.vimeo.com/videos/' + video_id, video_id, headers=headers)
+            for download_data in original_response.get('download'):
+                if download_data['quality'] == 'source':
+                    return {
+                        'url': download_data['link'],
+                        'ext': download_data['link'][-3:] or 'mp4',
```
Comment (Ashish0804) [inline yt_dlp/extractor/vimeo.py:265]: idk of any 4 letter extension for a video file.
Lemme know how this should be handled 
`determine_ext` this wasn't working

File: yt_dlp/extractor/vimeo.py (hunk 2 above)
Comment (pukkandan) [reply yt_dlp/extractor/vimeo.py:259]: I try not to use fstrings with nested quotes - they are hard to read and easy to make mistakes in. But if u must, I guess `'` in outer makes more sense with the rest of the code

Comment (Ashish0804) [reply yt_dlp/extractor/vimeo.py]: Neither was looking decent so i removed the f string.

File: yt_dlp/extractor/vimeo.py (hunk 3 above)
Comment (Ashish0804) [reply yt_dlp/extractor/vimeo.py]: This should be more robust now considering how the api gives url.

File: yt_dlp/extractor/vimeo.py (hunk 1 above)
Comment (pukkandan) [reply yt_dlp/extractor/vimeo.py]: why have seperate functions if you cant even think of seperate names for them

**Pull request merged**